examples/Example2/example2.cu

```diff
@@ -52,26 +204,62 @@ __global__ void DefinePhysicalStepLength(adept::BlockData<track> *block, process
   for (int i = blockIdx.x * blockDim.x + threadIdx.x; i < n; i += blockDim.x * gridDim.x) {
     // skip particles that are already dead
     track &mytrack = (*block)[i];
+
+    // Experimental limit on number of steps - to avoid 'forever' particles
+    constexpr uint16_t maxNumSteps = 250;  // Configurable in real simulation -- 1000 ?
+    ++  mytrack.num_step;
+    if( mytrack.num_step > maxNumSteps ) {
+       mytrack.status = dead;
+    }
+    
     if (mytrack.status == dead) continue;
 
     // return value (if step limited by physics or geometry) not used for the moment
     // now, I know which process wins, so I add the particle to the appropriate queue
     float physics_step = proclist->GetPhysicsInteractionLength(i, block);
-    float step         = LoopNavigator::ComputeStepAndPropagatedState(mytrack.pos, mytrack.dir, physics_step,
-                                                              mytrack.current_state, mytrack.next_state);
-    mytrack.pos += (step + kPush) * mytrack.dir;
-    if (mytrack.next_state.IsOnBoundary()) {
+    float step= 0.0;
+
+    if( ! BfieldOn ) {
```

Comment:
So this becomes now an embryonic transportation process. I guess, we should also modularize it and put into a separate class.

Reply:
Done - see above.